crates/common/tedge_config_macros/src/all_or_nothing.rs

```diff
@@ -0,0 +1,87 @@
+use crate::OptionalConfig;
+
+pub fn all_or_nothing<T, U>(
```

Comment:
At the moment I'm trying to keep this bit simple as this is an area where we have a lot of scope for solving problems we don't really have. I will refactor this to a single tuple argument via a trait so we at least don't need to create a different `all_or_nothing` function for each cardinality, but for now, I'm going to avoid coming up with any particularly clever solutions to generalising this, as I'm not convinced we really have a use case.